src/main/java/com/lginterview/exceptions/ApiExceptionHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.lginterview.exceptions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import javax.servlet.http.HttpServletResponse;
+
+@ControllerAdvice
+public class ApiExceptionHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(ApiExceptionHandler.class);
+
+    @ExceptionHandler(value = {CourseDataRetrievalException.class})
+    public void dataRetrievalException(CourseDataRetrievalException e, HttpServletResponse response) {
+        try {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        } catch (Exception se) {
+            logger.error("Exception sending error response:", se);
+        }
+    }
+
+    @ExceptionHandler(value = {IllegalArgumentException.class})
```

Comment:
[MS][Minor] Don't map IllegalArgumentException to HttpServletResponse.SC_BAD_REQUEST, because it most of the time indicates Server-Side implementation bug.

Even in your code, if someone would add a new RequestType.DELETE would throw IllegalArgumentException:
```
        if (courseActions.containsKey(courseRequest.getType()))
            return courseActions.get(courseRequest.getType()).apply(courseRequest);
        else
            throw new IllegalArgumentException("Invalid Course Request type");
```
and it would be actually implementation bug not a client error.